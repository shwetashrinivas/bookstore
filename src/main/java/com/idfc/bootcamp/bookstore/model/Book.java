package com.idfc.bootcamp.bookstore.model;

import jakarta.persistence.*;

@Entity
@Table(name="books")
public class Book {
    private String title;

    public Book(String title) {
        this.title = title;
    }

    @Id
    @GeneratedValue(strategy =  GenerationType.IDENTITY)
    private Long id;

    public Book(){}

    public String getTitle() {return title;}
}