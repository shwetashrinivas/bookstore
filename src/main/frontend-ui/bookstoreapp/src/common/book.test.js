import { render, screen } from "@testing-library/react";
import Book from "./book";

const book = {
  id: 1,
  title: "Tale of two cities",
  author: "Charles Dickens",
  price: 350.0,
  description:
    "The novel tells the story of the French Doctor Manette, his 18-year-long imprisonment in the Bastille in Paris, and his release to live in London with his daughter Lucie whom he had never met. The story is set against the conditions that led up to the French Revolution and the Reign of Terror.",
  coverImageUrl: "1586_f1601742134.png",
  rating: 5.0,
  quantity: 3,
};

describe("Book", () => {
  test("should rendering book component", () => {
    const { container } = render(<Book book={book} />);
    expect(container).toMatchSnapshot();
  });

  test("should have elements title, price, coverImageUrl, author", () => {
    render(<Book book={book} />);
    expect(screen.getByTestId("title")).toBeInTheDocument();
    expect(screen.getByTestId("price")).toBeInTheDocument();
    expect(screen.getByTestId("coverImageUrl")).toBeInTheDocument();
    expect(screen.getByTestId("author")).toBeInTheDocument();
  });
});
