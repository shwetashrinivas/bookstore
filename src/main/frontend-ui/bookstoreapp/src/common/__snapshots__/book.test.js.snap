// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should rendering book component 1`] = `
<div>
  <div
    class="book-component"
    data-testid="book"
  >
    <img
      alt="abc"
      data-testid="coverImageUrl"
      height="240"
      src="https://res.cloudinary.com/www-thepencilapp-com/image/upload/q_auto,f_auto/books/covers/1586_f1601742134.png"
      width="180"
    />
    <p
      class="title"
      data-testid="title"
    >
      Tale of two cities
    </p>
    <p
      class="author"
      data-testid="author"
    >
      by $
      Charles Dickens
    </p>
    <p
      class="price"
      data-testid="price"
    >
      RS $
      350
    </p>
  </div>
</div>
`;
